src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -177,6 +179,27 @@ public function getEnv($prefix, $name, \Closure $getEnv)
             return $env;
         }
 
+        if ('url' === $prefix) {
+            $env = \parse_url($env);
+
+            if (false === $env) {
+                throw new RuntimeException(sprintf('Invalid URL in env var "%s"', $name));
+            }
+
+            if (isset($env['path'])) {
+                // remove the '/' separator
+                $env['path'] = '/' === ($env['path'][0] ?? '') ? substr($env['path'], 1) : $env['path'];
```

Comment:
I really wonder if we shouldn't remove this line.

PRO: except for parsing a HTTP urls, this `/` is annoying (ie. when parsing a SGDB DSN we don't wont this `/` in the databse name), moreover we don't provides tools to remove it in the env processor.
CON: this is the default behavior of parse_url, users may except a path starting with a slash.

Shouldn't we provide a new key like `path_without_leading_slash` (please help me to find better name) ?
PRO: everybody is happy
CON: not REALLY the default behavior of parse_url

Reply:
But do we really need to have the same behavior of parse_url?

Reply:
removing the `/` is legit to me, the `:` is also removed from `host:port`
the thing is: should we allow relative URLs?
the behavior of parse_url makes sense to differentiate relative from absolute URLs.
Removing the `/` makes sense if we reject relative ones.
WDYT?

Reply:
added a new key `path_rootless` which is part of the RFC https://tools.ietf.org/html/rfc3986#section-3.3

Reply:
Wouldn't it make more sense to throw when scheme or host are not defined? then we can safely remove the leading `/` from path all the time.

Reply:
done.